zoomdls/src/main/java/com/zoomcar/uikit/optionsItem/ZSelectionItemCell.kt

```diff
@@ -0,0 +1,120 @@
+import com.zoomcar.util.isValid
+import com.zoomcar.zoomdls.R
+import com.zoomcar.zoomdls.databinding.LayoutSelectionItemBinding
+import kotlinx.android.parcel.Parcelize
+
+class ZSelectionItemCell : ConstraintLayout, View.OnClickListener {
+    private val binding: LayoutSelectionItemBinding
+    private var listener: IZSelectionItemListener? = null
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context,
+                attrs: AttributeSet?
+    ) : super(context, attrs)
+
+    constructor(context: Context,
+                attrs: AttributeSet? = null,
+                @AttrRes defStyleAttr: Int = 0
+    ) : super(context, attrs, defStyleAttr)
+
+    init {
+        val inflater: LayoutInflater = LayoutInflater.from(context)
+        binding = DataBindingUtil.inflate(inflater, R.layout.layout_selection_item, this, true)
+        binding.layoutSelectionItem.setOnClickListener(this)
+    }
+
+    fun setData(model: SelectionItemUIModel) {
+        binding.root.tag = model.id
+        binding.textEdit.apply {
+            isVisible = model.isEditable
+            setOnClickListener(this@ZSelectionItemCell)
+        }
+        when (model.type) {
+            SelectionItemType.DEFAULT -> {
+                binding.imageType.setImageResource(R.drawable.ic_circle_grey)
+                binding.textTitle.apply {
+                    TextViewCompat.setTextAppearance(this, R.style.OverlineSecondary);
+                }
+                binding.textDesc.apply {
+                    TextViewCompat.setTextAppearance(this, R.style.CaptionPrimary);
+                }
+            }
+            SelectionItemType.HIGHLIGHTED -> {
+                binding.imageType.setImageResource(R.drawable.ic_circle_green)
+                binding.textTitle.apply {
+                    TextViewCompat.setTextAppearance(this, R.style.Button2Primary);
+                }
+                binding.textDesc.apply {
+                    TextViewCompat.setTextAppearance(this, R.style.CaptionInactive);
+                }
+            }
+            SelectionItemType.DROP_OFF_LOCATION -> {
+                binding.imageType.setImageResource(R.drawable.ic_circle_grey)
+                binding.textTitle.apply {
+                    TextViewCompat.setTextAppearance(this, R.style.Body2Primary);
+                }
+                binding.textDesc.apply {
+                    TextViewCompat.setTextAppearance(this, R.style.CaptionInactive);
+                }
+            }
+        }
+        binding.textTitle.apply {
+            text = model.title
+            isVisible = model.title.isValid()
+        }
+        binding.textDesc.apply {
+            text = model.desc
+            isVisible = model.desc.isValid()
+        }
+    }
+
+    fun setListener(listener: IZSelectionItemListener) {
+        this.listener = listener
+    }
+
+    enum class SelectionItemType {
+        DEFAULT,
+        HIGHLIGHTED,
+        DROP_OFF_LOCATION
```

Comment:
Let's change the name of `DROP_OFF_LOCATION` to something else (DEFAULT AND HIGHLIGHTED sounds good). Sounds very product related.